doc/user/content/transform-data/temporal-filters.md

```diff
@@ -0,0 +1,368 @@
+
+The traditional (non-temporal) filter responds to `(data, time, diff)` triples by applying a predicate to `data`, and either dropping or retaining the triple based on what it sees. However, if we did that with a temporal predicate only at the moment we received the update, I guess using the current `mz_now()`, we wouldn't do the right thing at all. We might drop the record as being too early yet, oblivious to the fact that the record should re-appear in the future. Similarly, if the record should be removed in the future, evaluating the predicate ***now*** doesn't have the right effect.
+
+The temporal filter is somewhat less traditional than its non-temporal counterpart. Rather than drop or retain records ***right now***, it will schedule the insertion and deletion of records in the future.
+
+The temporal filter operator looks for predicates of the form
+
+```
+mz_now() CMP_OP EXPRESSION
+```
+
+where `CMP_OP` is a comparison operation other than `!=` (*i.e.* the operators `=`, `<`, `<=`, `>`, `>=` and things like `BETWEEN` that reduce to them) and `EXPRESSION` is an expression that does not contain `mz_now()`. Roughly, the expression is a function of `data`, and once we evaluate it we get a bound on `mz_now()`. If we have several comparisons, we end up with bounds, maybe lower and maybe upper, which describe an interval of time.
+
+An update `(data, time, diff)` normally takes effect at `time` and is in then effect indefinitely. However, we can narrow its range of time to `[lower, upper)` by transforming the input update into two output updates:
+
+```
+(data, max(time, lower), diff)
+(data, upper, -diff)
+```
+
+This change delays the insertion of `data` until at least `lower`, and schedules its deletion at `upper`.
+
+There are a variety of corner cases to double-check, mostly around what to do if a bound is absent, or if they cross (you can write it; we need to make sure it doesn't break). You'll want to double-check that the above makes sense when `diff` is negative (a deletion undoes the window its insertion would have introduced). We also need to update our query optimizer as filters can now do slightly weirder things than they could before, and it is less clear that you should use these filters ***e.g.*** to drive equi-join planning.
+
+But actually, the above is basically the implementation. The whole file comes in at around 300 lines, and that's with comments and a copyright header.
+
+## Appendix: Resource requirements
```

Comment:
So ... I'm thinking

- we apply your patch 3 (which adds this resource requirements section) to https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/transform-data/patterns/temporal-filters.md file
- In that existing file, in the second paragraph that ends with "When using temporal filters, Materialize must be prepared to retract updates in the near future and will need resources to maintain these retractions.", we just add "For details, see [Appendix](#...)".